DuckDuckGo/SettingsViewController.swift

```diff
@@ -57,11 +61,27 @@ class SettingsViewController: UITableViewController {
         super.viewWillAppear(animated)
         
         configureAutoClearCellAccessory()
+        configureHomePageCellAccessory()
+        migrateFavoritesIfNeeded()
+    }
+    
+    private func migrateFavoritesIfNeeded() {
```

Comment:
This ensures the favorites are not lost if the user changes the setting.  However, there's no going back to favourites, so the user will have to set them up again (using the edit functionality in bookmarks).